Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -860,61 +860,63 @@ class BrowserViewController: UIViewController {
-            firefoxHomeViewController.libraryPanelDelegate = self
-            firefoxHomeViewController.browserBarViewDelegate = self
-            self.firefoxHomeViewController = firefoxHomeViewController
-            addChild(firefoxHomeViewController)
-            view.addSubview(firefoxHomeViewController.view)
-            firefoxHomeViewController.didMove(toParent: self)
-            view.bringSubviewToFront(overKeyboardContainer)
+    /// Show the home page
+    /// - Parameter inline: Inline is true when the homepage is created from the tab tray, a long press
+    /// on the tab bar to open a new tab or by pressing the home page button on the tab bar. Inline is false when
+    /// it's the zero search page, aka when the home page is shown by clicking the url bar from a loaded web page.
+    func showHomepage(inline: Bool) {
+        if self.homepageViewController == nil {
+            createHomepage(inline: inline)
         }
 
         if self.readerModeBar != nil {
             hideReaderModeBar(animated: false)
         }
 
-        firefoxHomeViewController?.applyTheme()
+        homepageViewController?.applyTheme()
 
         // We have to run this animation, even if the view is already showing
         // because there may be a hide animation running and we want to be sure
         // to override its results.
         UIView.animate(withDuration: 0.2, animations: { () -> Void in
-            self.firefoxHomeViewController?.view.alpha = 1
+            self.homepageViewController?.view.alpha = 1
         }, completion: { finished in
-            if finished {
-                self.webViewContainer.accessibilityElementsHidden = true
-                UIAccessibility.post(notification: UIAccessibility.Notification.screenChanged, argument: nil)
-            }
+            guard finished else { return }
+            self.webViewContainer.accessibilityElementsHidden = true
+            UIAccessibility.post(notification: UIAccessibility.Notification.screenChanged, argument: nil)
         })
         view.setNeedsUpdateConstraints()
         urlBar.locationView.reloadButton.reloadButtonState = .disabled
     }
 
-    func hideFirefoxHome(completion: (() -> Void)? = nil) {
-        guard let firefoxHomeViewController = self.firefoxHomeViewController else { return }
+    /// Once the homepage is created, browserViewController keeps a reference to it, never setting it to nil during
+    /// an app session. The homepage can be nil in the case of a user having a Blank Page or custom URL as it's new tab and homepage
+    private func createHomepage(inline: Bool) {
+        // Firefox home page tracking i.e. being shown from awesomebar vs bottom right hamburger menu
+        let trackingValue: TelemetryWrapper.EventValue = inline ? .openHomeFromPhotonMenuButton : .openHomeFromAwesomebar
+        TelemetryWrapper.recordEvent(category: .action, method: .open, object: .firefoxHomepage, value: trackingValue, extras: nil)
```

Comment:
Will this cause problems if it is now called less often?

Reply:
Yeah I missed that. I guess we could move it to the `viewWillAppear` in the homepageViewController? Not sure where else this would make sense otherwise.

Reply:
You'll need a way to pass along if it's from the awesomebar or menu button, but if you can find any easy way to solve that then it's a sensible place to put it.
Otherwise you could move it up a layer in BVC to the showHomepage method instead.

Reply:
We already pass down the inline parameter, as the `isZeroSearch` param so maybe I would use that one 🤔

Reply:
Just committed a possible solution. Turns out viewWillAppear/viewDidAppear wasn't doing what we needed, so I went with adding recordHomepageAppeared/recordHomepageDisappeared methods that get's called from BVC. Let me know if this works!

Reply:
This is working well @lmarceau I tested it locally